commit 9c74cd4afbd3b5b6d90366b7dcb2a7befa8a37f2
Author: Thara Gopinath <thara.gopinath@linaro.org>
Date:   Tue Nov 27 17:43:11 2018 -0500

    thermal: Fix locking in cooling device sysfs update cur_state
    
    [ Upstream commit 68000a0d983f539c95ebe5dccd4f29535c7ac0af ]
    
    Sysfs interface to update cooling device cur_state does not
    currently holding cooling device lock sometimes leading to
    stale values in cur_state if getting updated simultanelously
    from user space and thermal framework. Adding the proper locking
    code fixes this issue.
    
    Signed-off-by: Thara Gopinath <thara.gopinath@linaro.org>
    Signed-off-by: Zhang Rui <rui.zhang@intel.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/thermal/thermal_sysfs.c b/drivers/thermal/thermal_sysfs.c
index 2241ceae7d7f..aa99edb4dff7 100644
--- a/drivers/thermal/thermal_sysfs.c
+++ b/drivers/thermal/thermal_sysfs.c
@@ -712,11 +712,14 @@ cur_state_store(struct device *dev, struct device_attribute *attr,
 	if ((long)state < 0)
 		return -EINVAL;
 
+	mutex_lock(&cdev->lock);
+
 	result = cdev->ops->set_cur_state(cdev, state);
-	if (result)
-		return result;
-	thermal_cooling_device_stats_update(cdev, state);
-	return count;
+	if (!result)
+		thermal_cooling_device_stats_update(cdev, state);
+
+	mutex_unlock(&cdev->lock);
+	return result ? result : count;
 }
 
 static struct device_attribute
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1078_9c74cd4afbd3b5b6d90366b7dcb2a7befa8a37f2.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1078_9c74cd4afbd3b5b6d90366b7dcb2a7befa8a37f2.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
